kite-data/kite-data-spark/src/test/scala/org/kitesdk/data/spark/SparkKiteSpec.scala

```diff
@@ -0,0 +1,305 @@
+/*
+ * Copyright 2015 Cloudera, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kitesdk.data.spark
+
+import org.apache.avro.generic.GenericData.Record
+import org.apache.avro.generic.{GenericRecord, GenericRecordBuilder}
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.{SparkConf, SparkContext}
+import org.junit.runner.RunWith
+import org.kitesdk.data._
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{BeforeAndAfterAll, MustMatchers, WordSpec}
+
+import scala.beans.BeanProperty
+import scala.util.Random
+
+case class Person(@BeanProperty name: String, @BeanProperty age: Int)
+
+case class User(name: String, creationDate: Long, favoriteColor: String)
+
+@RunWith(classOf[JUnitRunner])
+class SparkKiteSpec extends WordSpec with MustMatchers with BeforeAndAfterAll with TestSupport {
+
+  "Spark" must {
+    "be able to create a SchemaRDD/Dataframe from a kite parquet dataset" in {
+
+      cleanup()
+
+      val conf = new SparkConf().
+        setAppName("spark-kite-spec-test").
+        set("spark.serializer", "org.apache.spark.serializer.KryoSerializer").
+        setMaster("local[16]")
+
+      val sparkContext: SparkContext = new SparkContext(conf)
+
+      val products: Dataset[GenericRecord] = generateDataset(Formats.PARQUET, CompressionType.Snappy)
+
+      val sqlContext = new SQLContext(sparkContext)
+
+      val data = sqlContext.kiteDatasetFile(products)
+
+      data.registerTempTable("product")
+
+      val res = sqlContext.sql("select * from product where id < 10")
+
+      res.map(row => (row.getAs[String](0), row.getAs[Long](1))).collect() must be(
+        for {
+          i <- 1 to 9
+        } yield (s"product-$i", i.toLong)
+      )
+
+      sparkContext.stop()
+
+      cleanup()
+
+    }
+  }
+
+  "Spark" must {
+    "be able to create a SchemaRDD/Dataframe from a kite avro dataset" in {
```

Comment:
This test is basically the same as the previous one, except it uses Avro - can you factor out into a common method?